scripts/reborn_webui_v2_live_qa/google_auth_helpers.py

```
@@ -0,0 +1,823 @@
+    for path, contents in rows:
+        try:
+            payload = json.loads(contents)
+        except (TypeError, json.JSONDecodeError):
+            continue
+        challenge = payload.get("challenge") if isinstance(payload, dict) else None
+        if not isinstance(challenge, dict):
+            continue
+        authorization_url = str(challenge.get("authorization_url") or "")
+        if not authorization_url:
+            continue
+        parsed = urllib.parse.urlparse(authorization_url)
```

Ruff (0.15.18)

[error] Undefined name `urllib`

(F821)